src/components/Modal/Modal.jsx

```diff
@@ -22,20 +24,27 @@ class Modal extends Component {
             open: nextProps.open,
             type: nextProps.type,  
             background_color: nextProps.background_color,
-            blur: nextProps.blur 
+            blur: nextProps.blur ,
+            display_close_button: nextProps.display_close_button
         })
     }
 
     show = () => {
         document.addEventListener('click', this.handleClick);
-        document.body.classList.add('modal-visible');
+        if (this.state.type === 'entire-screen')
+            document.body.classList.add('modal-visible-entire-screen');
+        else
+            document.body.classList.add('modal-visible-in-place');
     }
 
     close = () => {
         this.setState({ open: false },() => {
             document.removeEventListener('click', this.handleClick);
           });
-        document.body.classList.remove('modal-visible');
+        if (this.state.type === 'entire-screen')
+            document.body.classList.remove('modal-visible-entire-screen');
+        else
+            document.body.classList.remove('modal-visible-in-place');
```

Comment:
Is modal-visible-in-place defined anywhere? If the aim is to enable scrolling for in-place modals, maybe we can get rid of the else part completely.